tests/e2e/raw_block_volume.go

```diff
@@ -596,4 +606,352 @@ var _ = ginkgo.Describe("raw block volume support", func() {
+		pod, err = createPod(client, namespace, nil, []*corev1.PersistentVolumeClaim{pvc}, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("Verify volume after expansion: %s is attached to the node: %s",
+			volumeID, pod.Spec.NodeName))
+		vmUUID = getNodeUUID(ctx, client, pod.Spec.NodeName)
+		if guestCluster {
+			vmUUID, err = getVMUUIDFromNodeName(pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		isDiskAttached, err = e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+		pvcConditions := pvc.Status.Conditions
+		expectEqual(len(pvcConditions), 0, "pvc should not have conditions")
+
+		ginkgo.By("Verify volume size after expansion")
+		blockDevSize, err := getBlockDevSizeInSectors(f, namespace, pod, devicePath)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// Filesystem size may be smaller than the size of the block volume.
+		// Here since filesystem was already formatted on the original volume,
+		// we can compare the new filesystem size with the original filesystem size.
+		if blockDevSize < originalBlockDevSize {
+			framework.Failf("error updating volume size for %q. Resulting volume size is %d", pvc.Name, blockDevSize)
+		}
+		ginkgo.By("Resized volume has been attached successfully")
+
+		// Verify original data on raw block volume after expansion
+		// Write data on expanded space to verify the expansion is sucessful and accessible.
+		ginkgo.By(fmt.Sprintf("Verify previously written data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		readDataFromRawBlockVolume(namespace, pod.Name, devicePath, "This is write before offline expansion")
+		ginkgo.By(fmt.Sprintf("Write random data of total size %v on raw volume at path %v inside pod %v",
+			volsizeInMiBAfterExpansion, pod.Spec.Containers[0].VolumeDevices[0].DevicePath, pod.Name))
+		writeDataOnRawBlockVolume(namespace, pod.Name, devicePath, "", volsizeInMiBAfterExpansion)
+
+		// Delete POD
+		ginkgo.By(fmt.Sprintf("Deleting the new pod %s in namespace %s after expansion", pod.Name, namespace))
+		err = fpod.DeletePodWithWait(client, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify volume is detached from the node after expansion")
+		isDiskDetached, err = e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskDetached).To(gomega.BeTrue(),
+			fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		if guestCluster {
+			ginkgo.By("Waiting for 30 seconds to allow CnsNodeVMAttachment controller to reconcile resource")
+			time.Sleep(waitTimeForCNSNodeVMAttachmentReconciler)
```

Comment:
we already have a wait call below why are sleeping additionally here

Reply:
This is waiting for reconcilation of CNSNodeVMAttachment CR to be completed on SV setup, then we check if the CR has been updated to reflect the detach of volume.